vterm.el

```diff
@@ -129,13 +129,23 @@ be send to the terminal."
                           :sentinel #'ignore))))
 
 ;; Keybindings
-(define-key vterm-mode-map [t] #'vterm--self-insert)
+;; (define-key vterm-mode-map [t] #'vterm--self-insert)
 (define-key vterm-mode-map [mouse-1] nil)
 (define-key vterm-mode-map [mouse-2] nil)
 (define-key vterm-mode-map [mouse-3] nil)
 (define-key vterm-mode-map [mouse-4] #'ignore)
 (define-key vterm-mode-map [mouse-5] #'ignore)
-(dolist (prefix '("M-" "C-"))
+(define-key vterm-mode-map [(tab)]  #'vterm--self-insert)
+(define-key vterm-mode-map (kbd "TAB")   #'vterm--self-insert)
+(define-key vterm-mode-map (kbd "<backspace>")   #'vterm--self-insert)
+(define-key vterm-mode-map (kbd "M-<backspace>")   #'vterm--self-insert)
+
+(defvar vterm-esc-map (make-keymap))
```

Comment:
What is the purpose of making this and binding it to `\e` ?

Reply:
Ok, I think I figured and I think it is to allow passing the `ESC` key to vterm (?). We should be careful that is does not conflict with `evil`, which relies heavily on the `ESC` key to change modes

Reply:
> Also, is there a reason why `make-keymap` instead of `make-sparse-keymap`? Conventional Emacs wisdom is to always use `make-sparse-keymap` unless you know for a fact you need `make-keymap`, which is almost never the case.

ok changed to  `make-sparse-keymap`

Reply:
I can't quit `htop` or `less` using ESC even in a normal terminal? (outside emacs).

Reply:
Ooooops my bad 😅

Reply:
I had a conflict with `evil` but it is because `evil` binds ESC for itself.

I feel like evil integration should be the concern of a separate PR.